cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -911,9 +924,15 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
         }
 
         Future<ReconciliationState> kafkaRollingUpdate() {
-            return withVoid(kafkaSetOperations.maybeRollingUpdate(kafkaDiffs.resource(), pod ->
-                isPodToRestart(kafkaDiffs.resource(), pod, kafkaAncillaryCmChange, this.clusterCa, this.clientsCa)
-            ));
+            return withVoid(kafkaSetOperations.maybeRollingUpdate(kafkaDiffs.resource(), pod -> {
+                boolean isPodToRestart = isPodToRestart(kafkaDiffs.resource(), pod, kafkaAncillaryCmChange, this.clusterCa, this.clientsCa);
```

Comment:
Another reason to do this is that `isPodToRestart()` does detailed logging about why we're rolling. It would be confusing to the reader of the log to see all that only for the decision to roll to be overruled by the maint window. If we need the ability to selectively disable the maint window stuff that's just an `honourMaintWindow` parameter to `isPodToRestart()`.

Reply:
Logging stuff is really a good point. The user could see messages like "Rolling pod .. due to ... " and then nothing happens. So it's worth moving checking windows inside.